src/test.rs

```diff
@@ -14,18 +13,12 @@ use rustc_serialize::{
 };
 
 use super::{
-    EncoderWriter,
-    DecoderReader,
     encode,
     decode,
-    SizeLimit
 };
 use super::SizeLimit::Infinite;
 
-fn the_same<'a,
-            V: Encodable<EncoderWriter<'a, MemWriter>, IoError>  +
-               Decodable<DecoderReader<'a, MemReader>, IoError> +
-               PartialEq + Show>(element: V) {
+fn the_same<'a, V>(element: V) where V: Encodable, V: Decodable, V: PartialEq, V: Show {
```

Comment:
I find the `where` notation nicer for long lists, and I believe the `+` syntax is sort of frowned upon.

In other words, no good reason :).